src/swap/components/SwapSettingsSlippageInput.tsx

```diff
@@ -12,10 +12,23 @@ export function SwapSettingsSlippageInput({
   className,
   defaultSlippage = 3,
 }: SwapSettingsSlippageInputReact) {
-  const { setLifeCycleStatus } = useSwapContext();
-  const [slippage, setSlippage] = useState(defaultSlippage);
+  const { setLifeCycleStatus, lifeCycleStatus } = useSwapContext();
+  const getMaxSlippage = () => {
+    if (
+      !!lifeCycleStatus.statusData &&
+      'maxSlippage' in lifeCycleStatus.statusData
+    ) {
+      return lifeCycleStatus.statusData.maxSlippage;
+    }
+    return defaultSlippage;
+  };
+
+  // TODO: Add comment about how we do these checks so the dropdown opens to the correct slippage value and button that it left off on
+  const [slippage, setSlippage] = useState(getMaxSlippage());
   const [slippageSetting, setSlippageSetting] = useState(
-    SLIPPAGE_SETTINGS.AUTO,
+    getMaxSlippage() === defaultSlippage
```

Comment:
I don't follow when this happens?

Reply:
This was to fix a bug when that dropdown was opened and closed it would always reopen to the state `Auto` which would then set the slippage to the default value of 3. 

This code checks if we have updated the slippage value already, if yes then we know that we are in the `CUSTOM` slippage_setting and reopen the dropdown to that state.

Reply:
ok, just know this condition `getMaxSlippage() === defaultSlippage` is literally fired at every re-render, which makes this an expensive computation.

You always want to think how to have `useEffect` to avoid those computation cost.

Reply:
Pretty much never run a function `potato()` in the component without making sure is bounded with useCallback or useEffect.

Otherwise the cost computation goes up

Reply:
Ah makes sense, updated to wrap getMaxSlippage with useCallback